lncfg/db.go

```diff
@@ -0,0 +1,60 @@
+package lncfg
+
+import (
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/channeldb/kvdb"
+)
+
+// DB holds database configuration for LND.
+type DB struct {
```

Comment:
What do you think about unpacking things a bit so we can spcify thing son the command line like: `--db.bolt.syncfreelist`? 

I think the the following structure would allow us to do so poretty easily: 
```go
type BoltDb struct {
    NoSyncFreeList bool `long:"nosyncfreelist`
}

type EtcdDb struct {
    <the rest of the fields as contained now>
}

type DB struct {
    Etcd *BoltDb `group:"bolt" namespace:"bolt"`

    Bolt *EtcdDb `group:"bolt" namespace:"bolt"`
}
```

Reply:
With the above, a user would specify `etcd` on the command line like `--db.etcd.host=x`, etc. When parsing these values, we'd ensure that only one of those internal pointers has been specified.

Reply:
I believe go-flags creates subgroups by default so pointer checking is not possible (although maybe possible to hack this together). So I instead created a top level `--db.backend` flag which by default is set to `bolt` and the sub configs proposed in your comment. PTAL.